File: 프로그래머스/2/42747. H－Index/H－Index.py
def solution(citations):
    answer = 0
    n = len(citations)
    
    citations.sort(reverse=True);
    
    for i in range(n):
        if i >= citations[i]:
            return i
    
    # h번 이상 인용된 논문이 h편이상, 나머지가 h번 이하 인용
    
    # citations 발표한 논문의 인용횟수 들이 담긴 배열
    # len(citations) 발표 논문 수
    
    return n

File: 프로그래머스/2/42747. H－Index/H－Index.js
function solution(citations) {
    let answer = 0;
    let n = citations.length;
    
    citations.sort((a, b) => b - a);    //내림차순 정렬
  
    // i    0 1 2 3 4
    // c[i] 6 5 3 1 0
    
    for (let i = 0; i < n; i++) {
        if (i >= citations[i]) {
            answer = i
            return answer
        }
    }
    
    return n
}